squad/api/rest.py

```diff
@@ -347,56 +348,46 @@ class Meta:
         fields = '__all__'
 
 
-class LatestTestResults(object):
-    def __init__(self, build, test_name):
-        if test_name is None:
-            raise serializers.ValidationError(_('"test_name" parameter is mandatory. Ex: suitename/testname'))
-
-        suite_and_test = test_name.rsplit("/", 1)
-        if len(suite_and_test) != 2:
-            raise serializers.ValidationError(_('"test_name" parameter should be in the format "suitename/testname"'))
-
-        self.build = build
-        self.test_runs = {tr.id: tr.environment for tr in self.build.test_runs.all()}
-        self.environments = build.project.environments.order_by("name", "slug")
-        self.test_suite_name, self.test_case_name = suite_and_test
-        self.test_list = Test.objects.filter(
-            test_run_id__in=self.test_runs.keys(),
-            name=self.test_case_name,
-            suite__slug=self.test_suite_name).prefetch_related('suite', 'known_issues')
+class LatestTestResultsSerializer(serializers.BaseSerializer):
+    def to_representation(self, build):
+        metadata = self.context.get('metadata')
+        project_environments = self.context.get('environments')
+        suite = self.context.get('suite')
 
+        test_runs = {tr.id: tr.environment for tr in build.test_runs.all()}
+        tests = Test.objects.filter(
+            test_run_id__in=test_runs.keys(),
+            metadata=metadata,
```

Comment:
So what if a test doesn't have metadata? as mentioned [here](https://github.com/Linaro/squad-client/issues/56#issuecomment-650328768), metadata is optional for tests.